.github/workflows/image.yml

```diff
@@ -0,0 +1,39 @@
+name: image
+on:
+  pull_request:
+    paths: [dockerfiles/Dockerfile.linux]
+  push:
+    branches: [main, test-me-*]
+    paths: [dockerfiles/Dockerfile.linux]
+
+jobs:
+  image:
+    strategy:
+      matrix:
+        arch: [x86_64, aarch64]
+    env:
+      TARGET_TAG: ghcr.io/getsentry/prebuilt-pythons-manylinux-${{ matrix.arch }}-ci
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: enable cross build
+      run: docker run --rm --privileged tonistiigi/binfmt --install arm64
+      if: matrix.arch == 'aarch64'
+    - name: build
+      run: |
+        args=()
+        if docker pull -q "$TARGET_TAG"; then
+          args+=(--cache-from "$TARGET_TAG")
+        fi
```

Comment:
Would you mind explaining how to read this?

Reply:
it's basically this but in bash, I hate bash: https://github.com/deadsnakes/runbooks/blob/db1267243c405d4f6c3b7780c89921fa510bf822/bin/build#L21

Reply:
I know this merge but I'm still confused about this whole block.

If pulling an image succeeds, why do we need the argument `--cache-from`?

I also don't see [Buildkit being enabled](https://docs.docker.com/engine/reference/builder/#buildkit) or `buildx` being used in order for [--cache-from to work](https://docs.docker.com/engine/reference/commandline/build/#specifying-external-cache-sources). I may be missing something.

I assume the remote image was built with `--build-arg BUILDKIT_INLINE_CACHE=1` and that's why you're intending to use `--cache-from` with it.

Reply:
`--cache-from` is a fairly old feature, predating buildkit by years

Reply:
Thanks for clarifying!